internal/provider/connection.go

```diff
@@ -107,6 +116,15 @@ func ConnectionFromResourceData(d *schema.ResourceData) (string, *ssh.ClientConf
 		clientConfig.Auth = append(clientConfig.Auth, ssh.PublicKeys(signer))
 	}
 
+	enable_agent, ok := d.GetOk("result_conn.0.agent")
+	if ok && enable_agent.(bool) {
+		sshAgent, err := net.Dial("unix", os.Getenv("SSH_AUTH_SOCK"))
+		if err != nil {
+			return "", nil, fmt.Errorf("couldn't connect to SSH agent: %s", err.Error())
+		}
+		clientConfig.Auth = append(clientConfig.Auth, ssh.PublicKeysCallback(agent.NewClient(sshAgent).Signers))
```

Comment:
It would be nice to have a test with `conn { agent = true }`. Could you attempt to add a test to `resource_remote_file_test.go`?

Reply:
Yes, definitely - my problem with a meaningful test atm is, that I am not sure yet whats the best way to add a SSH agent to the test suite here. I am not too familiar with go tooling tbh & my local docker setup doesn't support dns for container names, so I can't run it locally atm, only on GH.

Reply:
I think the ssh-agent may be added outside of go. It might be possible to simply start it before `go test` here: https://github.com/tenstad/terraform-provider-remote/blob/main/GNUmakefile#L13

It is an option to create  a new docker image for testing, based on `golang:1.16`, with everything needed for the agent to work.

Do you not get container name DNS, even when they are all in the same docker network (`docker network create remote`, `docker run --net remote`)? You can try `make testacc` and see if any tests pass to verify. If not, I can take a look at adding the test. If you are able to get an ssh-agent working in the `golang:1.16` image, against any host, it would be helpful.

Reply:
Got the tests working. Will add them in another PR after this one is merged.

https://github.com/tenstad/terraform-provider-remote/commit/af0a5511a147bd44e1f108854ab60372e488c0ee